### ISSUE TITLE
a contract suicide test case fail on ethermint

### ISSUE THREAD

**Issue by yihuang** [tharsis/ethermint#864]
__System info:__ ethermint main

__Steps to reproduce:__

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.6.6;

contract Inner {
    function destroy() public {
        selfdestruct(payable(msg.sender));
    }
}

contract Outer {
    function codesize_after_suicide(Inner inner) public {
        address addr = address(inner);
        inner.destroy();
        uint _size = 0;
        assembly {
            _size := extcodesize(addr)
        }
        require(_size > 0);
    }
}
```

- deploy `Inner` and `Outer`
- call `outer.codesize_after_suicide(inner)`, check the result
- it success on go-ethereum, but fail in ethermint.

__Expected behavior:__ [What you expected to happen]

__Actual behavior:__ [What actually happened]

__Additional info:__ [Include gist of relevant config, logs, etc.]

**Comment by yihuang** [comment]
BTW, it's fixed in [statedb-refactoring](https://github.com/tharsis/ethermint/pull/729), but maybe good to fix in main too, if the refactoring is not merged anytime soon.

**Comment by fedekunze** [comment]
we can add the test after #729 is merged

**Comment by crypto-facs** [comment]
I am going to close this issue as it was solved #729 was merged
